render_pdf/src/parser.py

```diff
@@ -0,0 +1,116 @@
+import json
+import sys
+from collections import defaultdict
+from typing import Optional
+from urllib.request import urlopen
+
+from lxml import etree
```

Comment:
see my earlier comments about using `xml.etree.ElementTree` instead of `lxml`. It has basically the same API with similar performance, but it isn't coupled to a library that we'll probably ditch in the future